ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/EthGetProof.java

```diff
@@ -0,0 +1,96 @@
+import tech.pegasys.pantheon.ethereum.core.BlockHeader;
+import tech.pegasys.pantheon.ethereum.core.MutableWorldState;
+import tech.pegasys.pantheon.ethereum.jsonrpc.RpcMethod;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.JsonRpcRequest;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.parameters.BlockParameter;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.parameters.JsonRpcParameter;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.queries.BlockchainQueries;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcError;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcErrorResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.results.proof.GetProofResult;
+import tech.pegasys.pantheon.ethereum.proof.WorldStateProof;
+import tech.pegasys.pantheon.util.bytes.Bytes32;
+import tech.pegasys.pantheon.util.bytes.BytesValue;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class EthGetProof implements JsonRpcMethod {
+
+  private final BlockchainQueries blockchain;
+  private final JsonRpcParameter parameters;
+
+  public EthGetProof(final BlockchainQueries blockchain, final JsonRpcParameter parameters) {
+    this.blockchain = blockchain;
+    this.parameters = parameters;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.ETH_GET_PROOF.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequest request) {
+
+    final Address address = parameters.required(request.getParams(), 0, Address.class);
+
+    final List<Bytes32> storageKeys =
+        Arrays.stream(parameters.required(request.getParams(), 1, String[].class))
+            .map(Bytes32::fromHexString)
+            .collect(Collectors.toList());
+
+    final long blockNumber =
+        resolveBlockNumber(parameters.required(request.getParams(), 2, BlockParameter.class));
+
+    final Optional<MutableWorldState> worldState = blockchain.getWorldState(blockNumber);
+
+    if (worldState.isPresent()) {
+      Optional<WorldStateProof<Bytes32, BytesValue>> proofOptional =
+          blockchain
+              .getWorldStateArchive()
+              .getWorldStateStorage()
+              .getAccountProof(worldState.get().rootHash(), address, storageKeys);
+      return proofOptional
+          .map(
+              proof ->
+                  (JsonRpcResponse)
+                      new JsonRpcSuccessResponse(
+                          request.getId(), GetProofResult.buildGetProofResult(address, proof)))
+          .orElse(new JsonRpcErrorResponse(request.getId(), JsonRpcError.NO_ACCOUNT_FOUND));
+    }
+
+    return new JsonRpcErrorResponse(request.getId(), JsonRpcError.INTERNAL_ERROR);
```

Comment:
It's not an internal error if the world state is unavailable.  Can you add a new error: `WORLD_STATE_UNAVAILABLE` ?